src/pds_doi_service/core/actions/test/util/email.py

```diff
@@ -0,0 +1,83 @@
+#
+#  Copyright 2022, by the California Institute of Technology.  ALL RIGHTS
+#  RESERVED. United States Government Sponsorship acknowledged. Any commercial
+#  use must be negotiated with the Office of Technology Transfer at the
+#  California Institute of Technology.
+#
+"""
+======
+email.py
+======
+
+Contains utility functions for testing email functionality.
+"""
+import configparser
+import os
+import signal
+import subprocess
+import tempfile
+import time
+from email.message import Message
+from email.parser import Parser
+from typing import Callable
+
+from pds_doi_service.core.util.config_parser import DOIConfigUtil
+
+
+def get_local_smtp_patched_config(self):
+    """
+    Return a modified default config that points to a local test smtp
+    server for use with the email test
+    """
+    parser = configparser.ConfigParser()
+
+    # default configuration
+    conf_default = "conf.ini.default"
+    conf_default_path = os.path.abspath(
+        os.path.join(os.path.dirname(__file__), os.pardir, os.pardir, os.pardir, "util", conf_default)
+    )
+
+    parser.read(conf_default_path)
+    parser["OTHER"]["emailer_local_host"] = "localhost"
+    parser["OTHER"]["emailer_port"] = "1025"
+
+    parser = DOIConfigUtil._resolve_relative_path(parser)
+
+    return parser
+
+
+def capture_email(f: Callable[[], None], port: int = 1025) -> Message:
+    """
+    Stand up a transient smtpd server, capture the first message sent through it, and return that message
+    :param f: a function which sends an email to the SMTP server
+    :param port: the port on which the sending process attempts to connect to the SMTP server
+    """
+    with tempfile.TemporaryFile() as temp_file:
+        # By default, all this server is does is echo email payloads to
+        # standard out, so provide a temp file to capture it
+        debug_email_proc = subprocess.Popen(
```

Comment:
Doesn't look like this is an issue (that it's using `python` rather than specifying an absolute path to a particular binary?).

Leaving this open in case someone wants to chime in.

Reply:
Forking a child Python should almost always use `sys.executable` so you get the same venv as the parent.

Reply:
Ah, roger that - will fix